tcms/testruns/static/testruns/js/get.js

```diff
@@ -210,3 +215,54 @@ function fileBugFromExecution(run_id, title_container, container, case_id, execu
 
     dialog.show();
 }
+
+function addLinkToExecution(testExecution) {
+
+    // remove all previous event handlers
+    $('.add-hyperlink-form').off('submit')
+
+    // this handler must be here, because if we bind it when the page is loaded.
+    // we have no way of knowing for what execution ID the form is submitted for.
+    $('.add-hyperlink-form').submit(() => {
+        const url = $('.add-hyperlink-form #id_url').val()
+        const name = $('.add-hyperlink-form #id_name').val()
+        const isDefect = $('.add-hyperlink-form #defectCheckbox').is(':checked')
+        const updateTracker = true
+
+        jsonRPC('TestExecution.add_link', [{
+            execution_id: testExecution.id,
+            url: url,
+            name: name,
+            is_defect: isDefect,
+        }, updateTracker], link => {
+            const ul = $(`.test-execution-${testExecution.id} .test-execution-hyperlinks`)
+            ul.append(renderLink(link))
+
+            // clean the values
+            $('.add-hyperlink-form #id_url').val('')
+            $('.add-hyperlink-form #id_name').val('')
+            $('.add-hyperlink-form #defectCheckbox').bootstrapSwitch('state', false)
+            $('.add-hyperlink-form #autoUpdateCheckbox').bootstrapSwitch('state', false)
+
+            // close the modal
+            $('#add-link-modal button.close').click()
+        })
+
+        return false;
+    })
+
+
+    return true; // so that the modal is opened
+}
+
+function renderLink(link) {
+    const linkEntryTemplate = $('#link-entry')[0].content
+    const template = $(linkEntryTemplate.cloneNode(true))
+    if (link.is_defect) {
+        template.find('.link-icon').addClass('fa fa-bug')
+    }
+
+    template.find('.link-url').html(link.name || link.url)
+
+    return template
+}
```

Comment:
btw this is way uglier, harder to read and if we benchmark it slower than html in js

Reply:
Performance is a non issue for UI actions initiated by the user. If this is ever so slow to the point that it will be noticeable by the user it will be because of the speed of the backend, not because how fast or slow the browser will render something.